package/kedro_viz/models/flowchart.py

```diff
@@ -496,189 +610,278 @@ def is_tracking_node(self):
+    )
+    stats: Optional[Dict] = Field(None, description="The statistics for the data node.")
+
+    @root_validator(pre=True)
+    def check_data_node_exists(cls, values):
+        assert "data_node" in values
+        cls.set_data_node_and_dataset(values["data_node"])
+        return values
+
+    @classmethod
+    def set_data_node_and_dataset(cls, data_node):
+        cls.data_node = data_node
+        cls.dataset = cast(AbstractDataset, data_node.kedro_obj)
 
         # dataset.release clears the cache before loading to ensure that this issue
         # does not arise: https://github.com/kedro-org/kedro-viz/pull/573.
-        dataset.release()
-        if not dataset.exists():
-            return
-
-        if data_node.is_plot_node():
-            self.plot = dataset.load()
-        elif data_node.is_image_node():
-            self.image = dataset.load()
-        elif data_node.is_tracking_node():
-            self.tracking_data = dataset.load()
-        elif data_node.is_preview_node():
+        cls.dataset.release()
+
+    @validator("type", always=True)
+    def set_type(cls, _):
+        return cls.data_node.dataset_type
+
+    @validator("filepath", always=True)
+    def set_filepath(cls, _):
+        dataset_description = cls.dataset._describe()
+        return _parse_filepath(dataset_description)
+
+    @validator("run_command", always=True)
+    def set_run_command(cls, _):
+        if not cls.data_node.is_free_input:
+            return f"kedro run --to-outputs={cls.data_node.name}"
+        return None
+
+    @validator("plot", always=True)
+    def set_plot(cls, _):
+        if cls.data_node.is_plot_node():
+            return cls.data_node.kedro_obj.load()
+        return None
+
+    @validator("image", always=True)
+    def set_image(cls, _):
+        if cls.data_node.is_image_node():
+            return cls.data_node.kedro_obj.load()
+        return None
+
+    @validator("tracking_data", always=True)
+    def set_tracking_data(cls, _):
+        if cls.data_node.is_tracking_node():
+            return cls.data_node.kedro_obj.load()
+        return None
+
+    @validator("preview", always=True)
+    def set_preview(cls, _):
+        if cls.data_node.is_preview_node():
             try:
-                if hasattr(dataset, "_preview"):
-                    self.preview = dataset._preview(**data_node.get_preview_args())
+                if hasattr(cls.dataset, "_preview"):
+                    return cls.dataset._preview(**cls.data_node.get_preview_args())
 
             except Exception as exc:  # pylint: disable=broad-except # pragma: no cover
                 logger.warning(
                     "'%s' could not be previewed. Full exception: %s: %s",
-                    data_node.name,
+                    cls.data_node.name,
                     type(exc).__name__,
                     exc,
                 )
+                return None
+        return None
```

Comment:
If you take a shortcut `if not cls.data_node.is_preview_node()` return `None` immediately, the logic will be less nested. (Github don't let me to add changes to the line you didn't edit)

Reply:
I agree, I was thinking of doing the same but I thought of sticking to the pattern of checking the true case (i.e., set_tracking_data, set_image functions above this one)